DuckDuckGo/TabViewController.swift

```diff
@@ -1742,6 +1758,8 @@ extension TabViewController: WKNavigationDelegate {
         }
 
         if navigationAction.isTargetingMainFrame(),
+           !navigationAction.isSameDocumentNavigation,
+           !navigationAction.shouldDownload,
```

Comment:
These are required changes for GPC protected sites not to break IFrame triggered downloads, as canceling and re-triggering request breaks the download attribute. This matches the logic on macOS where for page sub-requests we do not trigger the full distributed navigation chain execution. Aligned and accepted by the privacy team.